### ISSUE TITLE
Clicking Edit button opens wrong section for editing

### ISSUE THREAD

**Issue by whateverusernameisnottaken** [splitbrain/dokuwiki#3364]
When editing, if I click on the Edit button in a section, the editor opens the wrong section (typically the next later section below it) for editing. Strangely, if I click Cancel or Save, and then click Edit again for the same section, the editor will open the correct section for editing. 

I have no knowledge of, nor understanding of programming, so can't contribute anything in that sense.

**Comment by Klap-in** [comment]
Could you create a similar page at https://www.dokuwiki.org/playground:issue3364 ? Do you observe the same issue?
Which versions of the softwares are you using? Do add plugins to your wiki? Which? Do use another template?

**Comment by whateverusernameisnottaken** [comment]
I copied and pasted the text in, but noticed that it loses most of its formatting afterwards.  I have no administrative/server rights on the Wiki in question (wiki.FreshTomato.org). Is there some way I can get the version and plugin information you asked for without having to ask the Admin there? I've already asked him a lot of questions, as a newbie.

**Comment by Klap-in** [comment]
A lot of plugins can be listed with syntax of https://www.dokuwiki.org/plugin:info.

But I guess it is first more useful to explain bit better to us as outsider where you clicked, and what happened. Could you add a screenshot of the situation with notes on them where you clicked?

**Comment by whateverusernameisnottaken** [comment]
Sure, thanks. 

I'm working on this wiki page:
[](https://wiki.freshtomato.org/doku.php/network#ethernet_ports_state_-_configuration)

Let's say I click on the EDIT button in the WAN Settings section. If I do that, sometimes, the editor opens it. Sometimes, it opens the section below it, or two sections below it.

So instead of getting the WAN Settings section to edit, I get this:
![image](https://user-images.githubusercontent.com/21151880/103491270-50526900-4df0-11eb-9bb9-746cec3a3059.png)



As for the plugins, I don't think I can do that without administrative rights, correct?

**Comment by Klap-in** [comment]
You could, for dokuwiki.org you could see the plugins as well, you couldn’t?

I have not yet a clue, how it is possible.
I see a wysiwyg editor is used, maybe that added some extra steps behind the edit button? Not sure.

**Comment by whateverusernameisnottaken** [comment]
I have asked the system admin person to read through this thread. He's currently on holiday and won't be back for another week and a half. Sorry, I'm not clear what you meant when you wrote: "you could, for dokuwiki.org you could see the plugins as well"? How/in what menu would I see them ? Do you mean see a list of which ones are installed?

**Comment by Klap-in** [comment]
Installed plugins are listed in the extension plugin, which can be found in the admin menu.
https://www.dokuwiki.org/plugin:extension

**Comment by Klap-in** [comment]
What do you see if you hover your mouse over an Edit button? I expect that you see gray boxes around the text that will be edited with that specific button.

Please note that the button is placed below the section which it will load for editing.

I created also an account for this wiki, so far I cannot reproduce your issue, but I observe that the Edit button is easy associated with the heading near to them and at first glance less with text section above them, however if you hover you could discover.

**Comment by Michaelsy** [comment]
Related: https://forum.dokuwiki.org/d/18667-edit

\- Michael Sy.

**Comment by whateverusernameisnottaken** [comment]
Michaelsy: The thread to which you linked is also my thread.  When I hover over the edit (Section) button I see a balloon menu with the word "options" in it.

**Comment by Michaelsy** [comment]
> Michaelsy: The thread to which you linked is also my thread.

Yes I know that, that's why I cross linked. Please note that the benefit of forums is not primarily to help individual users, but rather to provide general help to the general public. 

For this reason, please also make the title of your DokuWiki forum thread more meaningful.

\- Michael Sy.

**Comment by whateverusernameisnottaken** [comment]
Michael: 

Thanks. I replied to your answer there, with further questions. I didn't know you could create a title thread. I still can't see how to do that. How do you do that?

**Comment by turnermm** [comment]
This user is using ckgedit.  It is difficult to see what may be happening from the supplied image.  But I suspect that there is another plugin that may be affecting the HTML and the resulting layout.  

Another observation.  The admin has opted to use one of two alternate GUIs for the editor, so it might be useful to restore the default editor GUI called moono to see what happens..   Unfortunately, I have no experience with the alternates, they are included as part of the CKEditor package.  But no one has every raised an issue with them before.

**Comment by whateverusernameisnottaken** [comment]
Okay, I'll look into those things (or pass them on to the admin-I have no admin access.)
He's on vacation, but he sent this list of installed plugins:

Access Counter and Popularity Plugin by HokkaidoPerson​
ACL Manager by Andreas Gohr
Plain Auth Plugin by Andreas Gohr​
BackupTool for DokuWiki 1.0.1 by Terence J. Grant​
CAPTCHA Plugin by Andreas Gohr​
ckgedit by Myron Turner​
Remove irrelevant entries from the change history by Dominik Eckelmann​
Configuration Manager by Christopher Smith​
denyactions plugin by Otto Vainio​
EditTable plugin by Andreas Gohr​
Extension Manager by Michael Hamann​
imgpaste plugin by Andreas Gohr​
Info Plugin by Andreas Gohr​
preregister by Myron Turner​
Publish Plugin by Jarrod Lowe, Andreas Gohr, Dominik Eckelmann​
Revert Manager by Andreas Gohr​
safefnrecode plugin by Andreas Gohr​
Searchindex Manager by Andreas Gohr​
sortablejs by vaxquis​
styling plugin by Andreas Gohr​
DokuWiki Upgrade Plugin by Andreas Gohr​
User History Manager Plugin by Ondra Zara, Rainbow-Spike​
User Manager by Chris Smith​

Are any of these known troublemakers?

**Comment by turnermm** [comment]
I've installed the alternate editor--which @whateverusernameisnottaken is using and haven't been able to reproduce this problem.  I suggest that this user should zip up one of the pages where this problem occurs an
[steingarten-Invoice 45229.pdf](https://github.com/splitbrain/dokuwiki/files/5797973/steingarten-Invoice.45229.pdf)
d email it to me at turnermm02Atshaw.ca

**Comment by turnermm** [comment]
I loaded the alternate editor into an instance of ckgedit and have not been able to reproduce this issue.  I suggest that @whateverusernameisnottaken zip up a few pages where this error occurs and attach it to this thread.  Then I can look at it and see what's happening.

**Comment by Klap-in** [comment]
I have made a login at the wiki, and I have tested a bit. So far I could not reproduce the situation as described. It could be that I use a different edit button than @whateverusernameisnottaken, or that other circumstances differ like: browser, operating system, extra user permission in the wiki, etc,etc. So probably @whateverusernameisnottaken needs to provide more guiding for us to enable to reproduce the case.

Some more general checks:
Did you cleared your browsers cache already? Use short cut `Ctrl+F5` please. Please try once to add `&purge=true` to the url.
Did you try once another computer and different browser?

**Comment by whateverusernameisnottaken** [comment]
I tried using EDGE. That made no difference. I don't have another PC on which I can try it right now, but maybe in a few days.

Windows 10 2004 Build 19041.685. 
Firefox 84.0.2
EDGE 87.0.664.75 (Official)

I strongly doubt that using another PC is going to change something. Clearing the cache using either method made no difference either. I will try the &purge=true next.

@ I suggest that @whateverusernameisnottaken zip up a few DokuWiki pages (i.e. not the HTML) where this error occurs and attach it to this thread. Then I can look at it and see what's happening.

Is something I can do with normal user rights, or do I need to ask an admin to do it?

**Comment by turnermm** [comment]
Just open the pages in the native DokuWiki editor, and copy and paste into notepad and save.

**Comment by whateverusernameisnottaken** [comment]
I already have the page saved as complete .html, would that be okay?

**Comment by turnermm** [comment]
> I already have the page saved as complete .html, would that be okay?

No.  I need the DokuWiki markup as it appears on the serve in the text file for the page.

**Comment by whateverusernameisnottaken** [comment]
Okay, here are the Basic/Network page and the Static DHCP/ARP/IPT pages from the wiki:


[Network and Static DHCP-ARP-IPT pages for github.zip](https://github.com/splitbrain/dokuwiki/files/5863727/Network.and.Static.DHCP-ARP-IPT.pages.for.github.zip)

**Comment by turnermm** [comment]
I've loaded these pages into both a linux and a windows 10 system.  The edit buttons worked correctly.  There were no issues.  

I did, however, notice something.  When I initially unzipped the file into a linux server, the pages would not open.   But when I copied the documents and pasted them into the editor, they were fine--they saved correctly and opened correctly.  Unzipped into  a Windows server, they loaded correctly and opened correctly.  The problem loading may have been thatthe pages as zipped up had Windows line endings CR/LF, although I'm not sure why that should have been a problem.

**Comment by whateverusernameisnottaken** [comment]
So, based on what you see, would that suggest this is a Dokuwiki install or server (Apache)? problem?

**Comment by turnermm** [comment]
Since I can't reproduce the problem, I can't tell.  It wouldn't be an apache issue.  Since it occurs independently in two different browsers, I would assume it's not a browser issue.  I don't see anything in the firefox console, so it doesn't appear to be an obvious javascript error.  I also, out of curiosity, switched to the alternate GUIs, and it makes no difference.  

I would try changing your template.  If it is not producing correct HTML, that could affect your editor buttons.

**Comment by Klap-in** [comment]
I have made a login at the wiki (wiki.FreshTomato.org), and I have tested a bit. So far I could not reproduce the situation as described..! Maybe I missed a step?

At which button should I press? Maybe mark at a screenshot which button I should use?

Do you know whether other users of the wiki experience also this issue? 
If no, please could you ask if other users of the wiki have the same issue?

**Comment by whateverusernameisnottaken** [comment]
Thanks Klap-in. The problem is intermittent. It doesn't always happen. It happens most often on the Basic/Network or Static DHCP/ARP page. 

I will ask on the Tomato forum if others are experiencing this issue.

What is a template?
